Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp

```diff
@@ -520,98 +541,87 @@ BOOST_AUTO_TEST_CASE(trackstate_reassignment) {
 
 BOOST_AUTO_TEST_CASE(storage_consistency) {
   MultiTrajectory<SourceLink> t;
+  size_t index = t.addTrackState();
+  auto ts = t.getTrackState(index);
+  auto [pc, fm] = make_trackstate(ts, TrackStatePropMask::All);
 
-  auto [ts, fm] = make_trackstate();
+  // now investigate the proxy
+  // parameters
+  BOOST_CHECK(ts.hasPredicted());
+  BOOST_CHECK_EQUAL(pc.predicted->parameters(), ts.predicted());
+  BOOST_CHECK_EQUAL(*pc.predicted->covariance(), ts.predictedCovariance());
 
-  // now put it into the collection
-  t.addTrackState(ts);
+  BOOST_CHECK(ts.hasFiltered());
+  BOOST_CHECK_EQUAL(pc.filtered->parameters(), ts.filtered());
+  BOOST_CHECK_EQUAL(*pc.filtered->covariance(), ts.filteredCovariance());
 
-  // now investigate the proxy
-  auto tsProxy = t.getTrackState(0);
+  BOOST_CHECK(ts.hasSmoothed());
+  BOOST_CHECK_EQUAL(pc.smoothed->parameters(), ts.smoothed());
+  BOOST_CHECK_EQUAL(*pc.smoothed->covariance(), ts.smoothedCovariance());
 
-  // parameters
-  BOOST_CHECK(tsProxy.hasPredicted());
-  BOOST_CHECK_EQUAL(ts.parameter.predicted->parameters(), tsProxy.predicted());
-  BOOST_CHECK_EQUAL(*ts.parameter.predicted->covariance(),
-                    tsProxy.predictedCovariance());
-
-  BOOST_CHECK(tsProxy.hasFiltered());
-  BOOST_CHECK_EQUAL(ts.parameter.filtered->parameters(), tsProxy.filtered());
-  BOOST_CHECK_EQUAL(*ts.parameter.filtered->covariance(),
-                    tsProxy.filteredCovariance());
-
-  BOOST_CHECK(tsProxy.hasSmoothed());
-  BOOST_CHECK_EQUAL(ts.parameter.smoothed->parameters(), tsProxy.smoothed());
-  BOOST_CHECK_EQUAL(*ts.parameter.smoothed->covariance(),
-                    tsProxy.smoothedCovariance());
-
-  BOOST_CHECK_EQUAL(&tsProxy.referenceSurface(), &ts.referenceSurface());
-
-  BOOST_CHECK(tsProxy.hasJacobian());
-  BOOST_CHECK_EQUAL(tsProxy.jacobian(), *ts.parameter.jacobian);
-
-  BOOST_CHECK(tsProxy.hasProjector());
-  std::visit(
```

Comment:
Does this go away completely, or am I reading the diff wrong?

Reply:
The diff is a bit weird. Since I use this struct to store the (explicit) measurements for comparison, I removed the `std::visit`